### DISCUSSION TITLE
App FAQ: AppScript not available in fresh installed SW6

### DISCUSSION THREAD

**Discussion by DirkPersky** [shopware/platform#3330] [App FAQ]
### What kind of question do you have about the app system?

General Setup

### Describe your question or feedback

He 

We're continuing with our app.
But it looks like ` import AppClient from 'src/service/app-client.service.ts'` is not available on fresh installs, with a compiled JS.

We uploaded our app to the store. 
Created a “test system” for approval, but the browser console show us:
![image](https://github.com/shopware/platform/assets/1736266/a6f71471-ba7d-4d24-ad58-7e6980c1db39)

This happens due to the n(8984) call which should load the appClient but it is not present.
If we copy your “app-client.service.ts” into our app and compile it again than it workes in your test-system.

> When will that be fixed? 
> We don't want a copyright issue.

OR

> How can we fix it?

**Comment by shyim** [comment]
Are you on the latest SW6 version and did you used the latest SW6 version to compile it? 🤔

**Reply by DirkPersky** [reply]
compiled with: 6.5.5.2 
On our own Server.

Test-System is from Shopware self (https://account.shopware.com/):
![image](https://github.com/shopware/platform/assets/1736266/9d1bee7f-079c-406d-89c8-6b844d50142f)

Greetings

**Reply by shyim** [reply]
Ahh I think it's because of tree shaking of webpack. Can you try this:

go to `vendor/shopware/storefront/Resources/app/storefront/src/main.js`

and add 
```js
import AppClient from 'src/service/app-client.service.ts'

window.Foo = AppClient;
```
and built again the Storefront

**Reply by DirkPersky** [reply]
Na, same issue.

In our dev-system after a "bin/build-storefront.sh" it still works.

If we fresh install SW6 and upload the APP, the issue exists.
If we than run "bin/build-storefront.sh" it disappears.


But at the moment i cant upload to account-shopware becouse the Code-Review trolls. It says:
```
Basic plugin analysis failed.

The xml element setup:secret is only for local development, please remove it. You can find your generated app secret on your extension detail page in the master data section. For more information see https://docs.shopware.com/en/shopware-platform-dev-en/app-system-guide/setup#authorisation
```
But the secret isn't exist, i dont have touched the `manifest.xml`:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<manifest xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/trunk/src/Core/Framework/App/Manifest/Schema/manifest-2.0.xsd">
    <meta>
        <name>webkonClickAndCollectApp</name>
        <label>Click and Collect App</label>
        <description>Click and Collect App by web-kon Internetagentur</description>
        <author>web-kon Internetagentur</author>
        <copyright>(c) by web-kon Internetagentur</copyright>
        <version>1.0.0</version>
        <icon>Resources/config/plugin.png</icon>
        <license>proprietary</license>
        <!-- Optional, Translatable, in this element you can describe the changes the shop owner needs to apply to his shops privacy policy, e.g. because you process personal information on an external server -->
        <privacyPolicyExtensions>
            This app processes following personal information on servers based in the EU:
            - Address information
            - Order positions
            - Order value
            - Customer ID
        </privacyPolicyExtensions>
    </meta>

    <permissions>
        <read>shipping_method</read>
        <read>payment_method</read>
        <read>product</read>
        <read>order</read>
        <read>customer</read>
        <read>system_config</read>
        <update>order</update>
    </permissions>

    <allowed-hosts>
        <host>sass-shopware.webkon-internetagentur.de</host>
    </allowed-hosts>

    <setup>
        <registrationUrl>https://sass-shopware.webkon-internetagentur.de/api/registration/webkonClickAndCollectApp</registrationUrl>
    </setup>

    <admin>
        <main-module source="https://sass-shopware.webkon-internetagentur.de/click-and-collect/admin" />
    </admin>

    <webhooks>
        <webhook name="webkon-cuc-order" url="https://sass-shopware.webkon-internetagentur.de/api/cuc/order" event="checkout.order.placed"/>
    </webhooks>

    <custom-fields>
        ....
    </custom-fields>
</manifest>

```

**Reply by shyim** [reply]
Maybe @tobiasberge knows something

**Reply by tobiasberge** [reply]
Sounds like the pre-build JS of the app is incompatible with the pre-build core JS. But whats strange is that the app-client.service already existed in `6.5.5.2`. Was the pre-compiled JS of the app (which is uploaded to the store) maybe created in an older version than `6.5.5.2`?

What happens when you import without the explicit `.ts` suffix?

**Reply by DirkPersky** [reply]
`What happens when you import without the explicit .ts suffix?` 
already tried last week didn't worked

`Was the pre-compiled JS of the app (which is uploaded to the store) maybe created in an older version than 6.5.5.2?`
Last week maybe on the first attempt. But then i updated and tryed it again without success.
Today with a fresh compield version too.

**Reply by tobiasberge** [reply]
With the .ts suffix I was just curios because we had a problem in the past with an extension alias in webpack (which basically mapped .js extensions to .ts extensions in case we migrated a file to .ts)

If the app JS was also build under 6.5.5.2, I cannot really spot the problem.
I will test it locally with 6.5.5.2 to write an App which uses the `app-client.service.ts` and see how it behaves.

**Reply by DirkPersky** [reply]
We updated our dev-system to "6.5.6.0" recompiled the storefront and uploaded it again to account.shopware.com.
But still same issue with your test-server.

![image](https://github.com/shopware/shopware/assets/1736266/e99dc6aa-214d-42b1-9ad9-b378e107ff28)

https://webkonclickandcollectapp-ruzjg.sw-testenvironment.de/shop/public/checkout/confirm

Searching for the "shopware-app-token" string in the all.js didnt get a resault:
![image](https://github.com/shopware/shopware/assets/1736266/0c88c128-2b8b-4b79-b4e7-dda01669b285)


app-client.service Line ~80
```js
   return {
            headers: {
                'shopware-app-token': data.token,
                'shopware-app-shop-id': data.shopId,
            },
            expires: data.expires,
        };
```

Through it looks like your AppClient need to pre compiled by default.
